demos/ret2spec_common.cc

```diff
@@ -0,0 +1,101 @@
+#include <array>
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+#include "cache_sidechannel.h"
+#include "instr.h"
+#include "ret2spec_common.h"
+#include "utils.h"
+
+// We cannot include the private_data twice, because that would lead to linking
+// error. Declaring it extern in here.
+extern const char *private_data;
+
+// Calls sched_yield in the cross-address-space version.
+void (*conditionally_unschedule)();
```

Comment:
Can we name this based on what it provides, rather than how we use it? They're the functions we call at the last level of recursion for `ReturnsTrue` and `ReturnsFalse`, so something like `returns_true_base_case` and `returns_false_base_case`?